src/vs/server/node/webClientServer.ts

```diff
@@ -148,7 +159,7 @@ export class WebClientServer {
 			return serveError(req, res, 400, `Bad request.`);
 		}
 
-		return serveFile(this._logService, req, res, filePath, headers);
+		return serveFile(filePath, this._environmentService.isBuilt ? CacheControl.NO_EXPIRY : CacheControl.NO_CACHING, this._logService, req, res, headers);
```

Comment:
When running from source, this should be `CacheControl.ETAG` to maintain existing behaviour.

Reply:
When running from sources I prefer no caching and I don't see a benefit of sticking to ETags. So unless you strongly object, I'd make that change.

Reply:
Let's continue in https://github.com/microsoft/vscode/issues/148642